pathgen/pathgen.go

```diff
@@ -1220,6 +1192,11 @@ func generateChildConstructorsForListBuilderFormat(methodBuf *strings.Builder, b
 		}
 	}
 
+	// Skip key builders for list with single key.
+	if keyN == 1 {
+		return errors
+	}
+
```

Comment:
I think it's not obvious to a raw reader why a single keyed list has special treatment, which is I think partially why you added a comment here. So I would add an explanation that a single-keyed list's non-builder methods are sufficient since either all or nothing is exactly the only ways you can construct its keys.

Reply:
changed to explaining why